### PR TITLE
Add docker file for aarch64 to install hdf5 dependency for h5py wheels

### PR DESCRIPTION
**Pull request by tbbharaj**
This PR addresses the comment on [PR-1924](https://github.com/h5py/h5py/pull/1924) to add Dockerfile for aarch64 to install hdf5 dependency for building h5py wheels.
Once merged, I will use this aarch64 image with installed hdf5 for building aarch64 wheels and update [PR-1924](https://github.com/h5py/h5py/pull/1924) 
  
If there are any questions/comments - I'd be happy fix things as needed.

Thank you
Tanveen

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks. For x86-64, we have a Github actions workflow that builds & publishes the container. Can we do something similar form aarch64, maybe using Travis again?

It doesn't necessarily need to publish to Github container registry - that's just a very convenient option when building on Github.

**Comment by takluyver** [comment]
I'm a bit torn. Building with emulation is really slow (~1 hour vs 6 minutes for the native build). But on the other hand, we don't have to change this very often, and it's convenient to have both images built on the same CI service. So maybe an hour each time it has to build is OK. @tacaswell @aragilar any preference?

You've also got a conflict you'll need to resolve before the PR can be merged - I think that's with my recent commit updating to HDF5 1.12.1.

**Comment by tbbharaj** [comment]
> I'm a bit torn. Building with emulation is really slow (~1 hour vs 6 minutes for the native build). But on the other hand, we don't have to change this very often, and it's convenient to have both images built on the same CI service. So maybe an hour each time it has to build is OK. @tacaswell @aragilar any preference?
> 
> You've also got a conflict you'll need to resolve before the PR can be merged - I think that's with my recent commit updating to HDF5 1.12.1.

Resolved merged conflict

**Comment by aragilar** [comment]
I'd express a preference for native building for the container, but if that's too complex, then go with the simpler option.

**Comment by takluyver** [comment]
I'd guess (without having tried it) that actually building it natively is easy enough, but it means another CI service on this repo, and another token to push it to a container registry. Github actions is convenient because Github also has a container registry, and each actions run automatically has a temporary token which is sufficient to push images to there.

**Comment by takluyver** [comment]
I've decided I'm OK with this using Github actions. The slowness of the emulated build is a bit ugly, but I think we change things rarely enough that it doesn't matter much practically, and I'd rather not have yet more CI to maintain.

If that proves to be a problem, we can always change it to another CI system at a later date.

File: .github/workflows/build-image.yml [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+name: Build Image from Dockerfile 
 on:
   push:
     branches: [ master ]
 
 jobs:
-  build:
-    name: Build image
+  build_x86_wheels:
+    name: Build image for x86
```
Comment (takluyver) [inline .github/workflows/build-image.yml:8]:
```suggestion
  build_x86_64_wheels:
    name: Build image for x86_64
```

x86 is generally used to mean the 32-bit architecture, which we're not doing at the moment.

File: Dockerfile_aarch64 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM quay.io/pypa/manylinux2014_aarch64
+
+ENV HDF5_VERSION 1.12.0
```
Comment (takluyver) [inline Dockerfile_aarch64:3]:
```suggestion
ENV HDF5_VERSION 1.12.1
```

**Review by takluyver — COMMENTED** [review]
Couple of minor changes

**Comment by tbbharaj** [comment]
Thank you for the feedbacks, are we good to merge this?
If there are any other concerns, I am happy to address!

**Comment by takluyver** [comment]
Thanks  @tbbharaj !

**Pull request merged**